src/app/modules/category/components/category-item/category-item.component.ts

```diff
@@ -0,0 +1,159 @@
+            viewValue: 'CATEGORY.WITH_REVIEWS',
+        },
+    ];
+
+    selected = CategoryConstants.popularValue;
+
+    readonly categoryProducts$ = this.store
+        .select(CategorySelectors.selectCategoryProducts)
+        .pipe(
+            map(data => data),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly categoryProductTitle$ = this.store
+        .select(CategorySelectors.selectCategoryTitle)
+        .pipe(
+            map(data => data),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly filteredDataSource$ = new BehaviorSubject<
+        Array<CategoryProductData>
+    >(null);
+
+    readonly filteredData$ = this.filteredDataSource$
+        .asObservable()
+        .pipe(takeUntil(this.unsubscribe$));
+
+    readonly isReadyToDisplay$ = combineLatest([
+        this.categoryProducts$,
+        this.categoryProductTitle$,
+    ]).pipe(
+        map(el => el.every(el => !!el)),
+        takeUntil(this.unsubscribe$),
+    );
+
+    constructor(
+        private readonly store: Store,
+        private readonly route: ActivatedRoute,
+    ) {}
+
+    ngOnInit(): void {
+        this.route.queryParams.subscribe((params: Params) => {
+            this.store.dispatch(
+                CategoryActions.getCategoryProductsById(params.id),
+            );
+        });
+        this.categoryProducts$.subscribe(data => {
+            this.filteredDataSource$.next(
+                this.sortFromMinToMax([...data], CategoryConstants.popular),
+            );
+        });
+    }
+
+    filterByValue(value: string): void {
+        this.categoryProducts$
+            .pipe(
+                take(1),
+                map(data => this.filterData(data, value)),
+            )
+            .subscribe(filteredData => {
+                this.filteredDataSource$.next(filteredData);
+            });
+    }
+
+    filterData(
+        data: Array<CategoryProductData>,
+        value: string,
+    ): Array<CategoryProductData> {
+        switch (data.length !== 0) {
+            case value === CategoryConstants.cheapValue:
+                return this.sortFromMinToMax(
+                    [...data],
+                    CategoryConstants.price,
+                );
+            case value === CategoryConstants.expensiveValue:
+                return this.sortFromMaxToMin(
+                    [...data],
+                    CategoryConstants.price,
+                );
+            case value === CategoryConstants.releaseDateValue:
+                return this.sortFromMaxToMin(
+                    [...data],
+                    CategoryConstants.releaseDate,
+                );
+            case value === CategoryConstants.popularValue:
+                return this.sortFromMinToMax(
+                    [...data],
+                    CategoryConstants.popular,
+                );
+            case value === CategoryConstants.reviewsValue:
+                return this.sortFromMaxToMin(
```

Comment:
create some variable and do 
someVar = CategoryConstants.reviews

Reply:
so then after switch statement i'll be able to call
 return this.sortFromMinToMax(
                    [...data],
                   someVar ,
                );